src/string_with_vars.ml

```diff
@@ -6,8 +6,12 @@ type item =
   | Text of string
   | Var of var_syntax * string
 
+type items =
+  | Unquoted_var of var_syntax * string
+  | Concat of item list
+
 type t =
```

Comment:
Personally, I would use:

```ocaml
type t =
  | Unquoted_var of Loc.t * var_syntax * string
  | Concat of Loc.t * item list
```

since the code is about the same and the memory usage is slightly better.